Marlin/src/pins/pins_MKS_13.h

```diff
@@ -53,6 +53,20 @@
 
 #include "pins_RAMPS.h"
 
+#ifdef Z_MIN_PROBE_ENDSTOP
+  #undef Z_MIN_PROBE_ENDSTOP
+#endif
+
+#if ENABLED(Z_MIN_PROBE_ENDSTOP)
+  #if  ENABLED(USE_ZMIN_PLUG)
+    #define Z_MAX_PIN -1
+    #define Z_MIN_PROBE_PIN 19
+  #elif ENABLED(USE_ZMAX_PLUG)
+    #define Z_MIN_PIN -1
+    #define Z_MIN_PROBE_PIN 18
+  #endif
+#endif
+
```

Comment:
I see some issues. First, `#ifdef`=>`#undef` is pointless. Just use `#undef`.

Second, since you just did `#undef Z_MIN_PROBE_ENDSTOP` the next line `#if ENABLED(Z_MIN_PROBE_ENDSTOP)` will always fail.

Third, the `USE_ZMIN_PLUG` and `USE_ZMAX_PLUG` options may both be in use for various reasons. They only tell Marlin that these plugs are connected to _**some**_ kind of endstop or probe. For example, a machine may have both Z max and Z min endstops, with the probe connected to the `X_MAX_PIN`.

It's not appropriate to modify pins assignments based on configuration options in this way. So, what's the right way to do what you're trying to do? Leave it as it was, I would say. The `Z_MIN_PROBE_PIN` is assigned by default to the `X_MAX_PIN`, and if users would prefer to use a different pin the `Configuration.h` instructions already explain how to set the `Z_MIN_PROBE_PIN` to something else.

Reply:
@thinkyhead Ok. How to set it to Z_MAX pin in Configuration.h? And where is instructions about it? Configuration.h only states this:
Since MKS Gen board uses ramps like pin assignment pins_MKS_13.h uses D32 pin (but i dont know location of this pin on board...) i want to use Z_MAX in my case. So how to set it to ZMAX?

```
/**
 * Z_MIN_PROBE_ENDSTOP
 *
 * Enable this option for a probe connected to any pin except Z-Min.
 * (By default Marlin assumes the Z-Max endstop pin.)
 * To use a custom Z Probe pin, set Z_MIN_PROBE_PIN below.
 *
 *  - The simplest option is to use a free endstop connector.
 *  - Use 5V for powered (usually inductive) sensors.
 *
 *  - RAMPS 1.3/1.4 boards may use the 5V, GND, and Aux4->D32 pin:
 *    - For simple switches connect...
 *      - normally-closed switches to GND and D32.
 *      - normally-open switches to 5V and D32.
 *
 * WARNING: Setting the wrong pin may have unexpected and potentially
 * disastrous consequences. Use with caution and do your homework.
 *
 */

```